test/s3_test.go

```diff
@@ -0,0 +1,255 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/require"
+)
+
+const output_bucket = "test-bucket"
+
+// initTestCases returns a list of BucketTestCase to be used for tests.
+func initTestCases() []BucketTestCase {
+	return []BucketTestCase{
+		{
+			testName: "TestBucket1",
+			vars: map[string]interface{}{
+				"test_bucket_name": "",
+				"read_only_paths":  []string{"path/to/ro-folder"},
+				"read_write_paths": []string{"path/to/rw-folder"},
+			},
+			objTestCases: []ObjectTestCase{
+				{
+					key:             "path/to/ro-folder/obj1",
+					encryption:      "AES256",
+					expectPassRead:  true,
+					expectPassWrite: false,
+				},
+				{
+					key:             "path/to/rw-folder/obj2",
+					encryption:      "AES256",
+					expectPassRead:  true,
+					expectPassWrite: true,
+				},
+				{
+					key:             "other/folder/obj3",
+					encryption:      "AES256",
+					expectPassRead:  false,
+					expectPassWrite: false,
+				},
+			},
+		},
+		{
+			testName: "TestBucket2",
+			vars: map[string]interface{}{
+				"test_bucket_name": "",
+				"read_only_paths":  []string{"other/path/to/ro-folder"},
+				"read_write_paths": []string{"other/path/to/rw-folder"},
+			},
```

Comment:
The idea for creating more buckets and "duplicating" the test was to just verify and create policies using different RO and RW folders and make sure they are created following the parameter. (instead of a potential non-wanted hardcoded `path/to/rw-folder`).
Having this list of test cases makes it easy to add more, so feel free to chime in with suggestions :)
I actually had a longer list while doing my local tests.. realized they were "non necessary duplicates" and didn't commit.

Reply:
Ok! I will update the second case to test multiple paths instead.

Reply:
done in https://github.com/hbarros-caylent/terraform-aws-s3/commit/6154dd2e03320d7b463d237590a6623eb0693159